src/objects/zcl_abapgit_objects.clas.abap

```diff
@@ -1268,4 +1278,72 @@ CLASS zcl_abapgit_objects IMPLEMENTATION.
     ENDLOOP.
 
   ENDMETHOD.
+
+
+  METHOD confirm_transport_message.
+
+    TYPES: BEGIN OF ty_s_message,
+             id TYPE symsgid,
+             ty TYPE symsgty,
+             no TYPE symsgno,
+             v1 TYPE symsgv,
+             v2 TYPE symsgv,
+             v3 TYPE symsgv,
+             v4 TYPE symsgv,
+           END OF ty_s_message.
+
+    DATA ls_message TYPE ty_s_message.
+
+    FIELD-SYMBOLS: <lt_confirmed_messages> TYPE STANDARD TABLE.
+
+    IF gv_confirm_trans_msg_possible = zif_abapgit_definitions=>c_no.
+      " auto-confirmation of transport messages is not possible in this system.
+      RETURN. ">>>>>>>>>
+    ENDIF.
+
+    IF gv_confirm_trans_msg_possible IS INITIAL.
+      " Auto-confirm certain messages (requires SAP Note 1609940)
+      PERFORM dummy IN PROGRAM saplstrd IF FOUND.  "load function group STRD once into memory
+    ENDIF.
+
+    ASSIGN ('(SAPLSTRD)GT_CONFIRMED_MESSAGES') TO <lt_confirmed_messages>.
+
+    IF sy-subrc = 0.
+      gv_confirm_trans_msg_possible = zif_abapgit_definitions=>c_yes.
+    ELSE.
+      gv_confirm_trans_msg_possible = zif_abapgit_definitions=>c_no.
+      RETURN. ">>>>>>>>>>>>>>
+    ENDIF.
+
+    ls_message-id = iv_msgid.
+    ls_message-no = iv_msgno.
+    COLLECT ls_message INTO <lt_confirmed_messages>.
```

Comment:
Does collect provide any benefit here compared to insert / append?

Reply:
It avoids duplicates in the table and is them same statment used by @mbtools so i just reused that. But insert should also be fine. Should i change? I don't like using append because it does not work for all table types. 
https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#prefer-insert-into-table-to-append-to

Reply:
I think in this case it does not COLLECT / sum anything, so INSERT is easier to understand, as the intention is just to insert, not sum/collect anything

Reply:
ok i change